controllers/yawol-controller/loadbalancerset/loadbalancerset_controller.go

```diff
@@ -175,39 +175,49 @@ func (r *LoadBalancerSetReconciler) reconcileStatus(
 func (r *LoadBalancerSetReconciler) reconcileReplicas(
 	ctx context.Context,
 	set *yawolv1beta1.LoadBalancerSet,
-	deletedMachines, notReadyMachines, readyMachines []yawolv1beta1.LoadBalancerMachine,
+	machines []yawolv1beta1.LoadBalancerMachine,
+	deletedMachineCount int,
 ) (ctrl.Result, error) {
-	var currentReplicas, desiredReplicas, deletedReplicas int
-	desiredReplicas = set.Spec.Replicas
-	currentReplicas = len(deletedMachines) + len(notReadyMachines) + len(readyMachines)
-	deletedReplicas = len(deletedMachines)
-
-	if currentReplicas < desiredReplicas {
+	if len(machines) < set.Spec.Replicas {
 		if err := r.createMachine(ctx, set); err != nil {
 			return ctrl.Result{}, err
 		}
 		return ctrl.Result{RequeueAfter: time.Second * 2}, nil
 	}
 
-	if desiredReplicas < currentReplicas-deletedReplicas {
-		// Delete not ready machines first
-		if len(notReadyMachines) > 0 {
-			if err := r.deleteMachine(ctx, &notReadyMachines[0]); err != nil {
-				return ctrl.Result{}, err
-			}
-			return ctrl.Result{RequeueAfter: time.Second * 2}, nil
+	if len(machines) < len(machines)-deletedMachineCount {
```

Comment:
shouldnt this be something like: len(machines) > desiredreplicas ? i don't quite understand why the deletedmachinecount is important for this calculation